fallbear-cmd

```diff
@@ -2,63 +2,85 @@
 
 echo_debug() {
     if [ -n "$PVDEBUG" ]; then
-	echo $@
+        echo "$@" >&2
     fi
 }
 
 lxc_pid() {
     target=$1
-    nsenter -m/proc/1/ns/mnt \
-	    lxc-info $target | grep PID: | sed -e 's/^[^0-9]*//'
+    nsenter --target 1 --mount -- lxc-info --no-humanize --pid "$target"
 }
 
 target=$USER
 p=1
 plat=${USER##*@}
 plat_tty=${USER%@$plat}
 
-
 if [ "$plat" = "$plat_tty" ]; then
-	plat_tty=
+    plat_tty=
 fi
 
-if ! [ "$plat" = "/" -o "$plat" = "_pv_"  ]; then
-    p=`lxc_pid $plat`
+case "$plat" in
+"/"|"_pv_")
+    # always unset plat_tty for pv env
+    plat_tty=
+    ;;
+*)
+    p="$(lxc_pid "$plat")"
 
     if [ -z "$p" ]; then
-	echo "no container found: $plat"
-	exit 1
+        echo "no container found: $plat" >&2
+        exit 1
     fi
-else
-	# always unset plat_tty for pv env
-	plat_tty=
-fi
+    ;;
+esac
 
 # if we try to go for a tty, we use lxc-console ...
 if [ -n "$plat_tty" ]; then
-	lxc-console --name $plat -t $plat_tty
-	exit $?
+    exec lxc-console --name "$plat" -t "$plat_tty"
 fi
 
 PVUSER=${PVUSER:-root}
-echo_debug PVUSER: $PVUSER
-pwdline=`nsenter -m/proc/$p/ns/mnt -- [ -f /etc/passwd ] \
-		 && (nsenter -m/proc/$p/ns/mnt cat /etc/passwd | grep ^$PVUSER:) \
-		 || true`
+echo_debug "PVUSER: $PVUSER"
+
+# get home directory and login shell from passwd entry, format is:
+# name:password:UID:GID:GECOS:directory:shell
+IFS=":" read -r _ _ _ _ _ home shell 2>/dev/null << EOF
+$(nsenter --target "$p" --mount -- grep "^$PVUSER:" /etc/passwd)
+EOF
+
+echo_debug "shell: $shell"
+echo_debug "home: $home"
```

Comment:
maybe move these one below so we see the actual final values used if debug is on.